### PR TITLE
Pvf refactor execute worker errors follow up

### PR DESCRIPTION
**Pull request by maksimryndin**
follow up of https://github.com/paritytech/polkadot-sdk/pull/2604
closes https://github.com/paritytech/polkadot-sdk/pull/2604

- [x] take relevant changes from Marcin's PR 
- [x] extract common duplicate code for workers (low-hanging fruits)

~Some unpassed ci problems are more general and should be fixed in master (see https://github.com/paritytech/polkadot-sdk/pull/4074)~

Proposed labels: **T0-node**, **R0-silent**, **I4-refactor**

-----

kusama address: FZXVQLqLbFV2otNXs6BMnNch54CFJ1idpWwjMb3Z8fTLQC6

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5866944

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5867647

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5872817

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5878120

**Review by s0me0ne-unkn0wn — APPROVED** [review]
This looks great, thanks a lot!

**Comment by s0me0ne-unkn0wn** [comment]
Oh you don't say

File: polkadot/node/core/pvf/common/src/error.rs [hunk 1]
```diff
@@ -136,6 +136,8 @@ pub enum InternalValidationError {
 	/// Could not find or open compiled artifact file.
 	#[error("validation: could not find or open compiled artifact file: {0}")]
 	CouldNotOpenFile(String),
+	#[error("validation: could not create pipe: {0}")]
```
[the remark this answers is not in the document]
Comment (maksimryndin) [reply polkadot/node/core/pvf/common/src/error.rs:140]: Thank you!) Fixed https://github.com/paritytech/polkadot-sdk/pull/4071/commits/5bdda49a518dcd50ececae259b3bd47709b83ac5

**Review by AndreiEres — APPROVED** [review]
Good!

File: polkadot/node/core/pvf/execute-worker/src/lib.rs [hunk 2]
```diff
@@ -127,13 +137,23 @@ pub fn worker_entrypoint(
 		|mut stream, worker_info, security_status| {
 			let artifact_path = worker_dir::execute_artifact(&worker_info.worker_dir_path);
 
-			let Handshake { executor_params } = recv_execute_handshake(&mut stream)?;
+			let Handshake { executor_params } = map_err!(
```
[the remark this answers is not in the document]
Comment (maksimryndin) [reply polkadot/node/core/pvf/execute-worker/src/lib.rs]: Sure) thanks:) refactored at a05897ff2665c7776b7dba6ab8b249f7f62c4177

File: polkadot/node/core/pvf/Cargo.toml [hunk 3]
```diff
@@ -17,16 +17,17 @@ cfg-if = "1.0"
...
+is_executable = { version = "1.0.1", optional = true }
 pin-project = "1.0.9"
 rand = "0.8.5"
 slotmap = "1.0"
 tempfile = "3.3.0"
 thiserror = { workspace = true }
 tokio = { version = "1.24.2", features = ["fs", "process"] }
 
-parity-scale-codec = { version = "3.6.1", default-features = false, features = ["derive"] }
+parity-scale-codec = { version = "3.6.1", default-features = false, features = [
```
[the remark this answers is not in the document]
Comment (maksimryndin) [reply polkadot/node/core/pvf/Cargo.toml:28]: Not sure if I've got it right, but it is the result of applying `taplo fmt polkadot/node/core/pvf/Cargo.toml --config .config/taplo.toml` from the ci job

**Pull request merged**